src/sql/azdata.proposed.d.ts

```diff
@@ -3321,8 +3334,16 @@ declare module 'azdata' {
 		onRowSelected: vscode.Event<any>;
 	}
 
+	export interface IRowCheckboxChangedArg {
+		checked: boolean;
+		row: number;
+		column: number;
+		name: string;
+	}
+
 	export interface TableComponent extends Component, TableComponentProperties {
 		onRowSelected: vscode.Event<any>;
+		onCheckBoxChanged?: vscode.Event<IRowCheckboxChangedArg>;
```

Comment:
onCellAction?:vscode.Event<ICellActionEventArgs>

interface ICellActionEventArgs {
    row: number;
    column: number;
    columnName: number;
}

button and checkbox event args can inherit from this interface